flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/HybridSourceReader.java

```diff
@@ -252,6 +252,8 @@ private void setCurrentReader(int index) {
     @Override
     public void pauseOrResumeSplits(
             Collection<String> splitsToPause, Collection<String> splitsToResume) {
-        currentReader.pauseOrResumeSplits(splitsToPause, splitsToResume);
+        if (currentReader != null) {
```

Comment:
if it is null, should we error? 
Would it fail later if we do not error here ?
Can we get an empty collection here - that we could also check for?

Reply:
Yes. Actually we had used the version without the null check for several months internally and didn't hit any issues. It implies the null check could be redundant. However, potentially more code will use this method in the future. So adding the check would be a defensive measure (I actually forgot to push this commit to my previous PR). I'm fine with both.